go/pkg/bertyprotocol/api_event.go

```diff
@@ -176,3 +177,36 @@ func (s *service) GroupMessageList(req *bertytypes.GroupMessageList_Request, sub
 
 	return nil
 }
+
+func (s *service) WaitMemberIsReady(req *bertytypes.WaitMemberIsReady_Request, sub ProtocolService_WaitMemberIsReadyServer) error {
+	groupPK, err := crypto.UnmarshalEd25519PublicKey(req.GroupPK)
+	if err != nil {
+		return errcode.ErrInvalidInput.Wrap(err)
+	}
+	memberPK, err := crypto.UnmarshalEd25519PublicKey(req.MemberPK)
+	if err != nil {
+		return errcode.ErrInvalidInput.Wrap(err)
+	}
+
+	id, err := groupPK.Raw()
+	if err != nil {
+		return errcode.ErrSerialization.Wrap(err)
+	}
+	cg, err := s.getContextGroupForID(id)
+	if err != nil {
+		return errcode.ErrGroupMissing.Wrap(err)
+	}
+
+	err = cg.MetadataStore().WaitSecretsAreSent(sub.Context(), memberPK)
```

Comment:
WaitSecretsAreSent() is blocking until all secrets are sent to members. if error happen, it will be stopping until sub's context is cancel. But WaitMemberIsReady() normally is called outside via grpc. I'm not sure we have timeout for gprc handler or not. Do you think in request should have a timeout value to control waiting duration?
for examples, after a timeout, we can try to do a sent missed secrets?

Reply:
`if error happen, it will be stopping until sub's context is cancel`
means that if any abnormal happen, like secret was not sent. so WaitSecretsAreSent() will block forever on waiting data from channel `chanReady`.
And I don't see any control over `sub.ctx` - so this function can block forever.
I'm also not quite familiar with GRPC.
@moul : do you have any idea here?

Reply:
The client can call the grpc method with a context.WithTimeout here to cancel : https://github.com/berty/berty/blob/bdd1babf29e06548309e6a125672b17e80bb50c4/go/pkg/bertyprotocol/scenario_test.go#L816

Reply:
This context is from client side of GRPC, and WaitMemberIsReady() will be called in server side. I'm not sure ctx of client side impact to server side. :).

Reply:
not sure about the whole discussion, but a thing that is sure, is that we can completely setup a timeout on gRPC, but it may have strange behavior, I prefer each call to have it's own limit so it can decide by itself what is a normal limit and what is not

Reply:
but anw, I think it's fine now. Thanks

Reply:
@moul:
In client side we call WaitMemberIsReady() via GRPC. and in server side, WaitMemberIsReady() will wait forever on `readyChannel` now or until context of handler is done - so we not sure the context lifetime of client side has any impact to context created in handler of server side. That is context I'm discussing.

Reply:
Resolved because this code will be removed, see https://github.com/berty/berty/pull/2312#discussion_r501000681